cycledash/genotypes.py

```diff
@@ -383,8 +383,31 @@ def _header_spec(vcf_header_text, extant_cols):
         path=['sample_name']) # This path is not the default super -> sub column
 
     # Add Cycledash-derived columns
-    column_name = 'annotations:gene_names'
-    if column_name in extant_cols:
+    _add_extant_column_to_spec(extant_cols, 'annotations:gene_names', res, 'String', 1,
+            ('Names of genes that overlap with this variant\'s '
+            'starting position, derived from Ensembl Release 75.'))
+    _add_extant_column_to_spec(extant_cols, 'annotations:varcode_gene_name', res, 'String', 1,
+            ('Name of the gene that overlaps with this variant\'s effect'
+            'derived from Varcode project.'))
+    _add_extant_column_to_spec(extant_cols, 'annotations:varcode_transcript', res, 'String', 1,
+            ('Transcript that overlaps with this variant'
+            'derived from Varcode project.'))
+    _add_extant_column_to_spec(extant_cols, 'annotations:varcode_effect_notation', res, 'String', 1,
+            ('Protein change caused by this variant, '
+            'derived from Varcode project.'))
+    _add_extant_column_to_spec(extant_cols, 'annotations:varcode_effect_type', res, 'String', 1,
+            ('Type of the variant, '
+            'derived from Varcode project.'))
+
+    # Remove empty supercolumns
+    for key, val in res.iteritems():
+        if not val.keys():
+            del res[key]
+
+    return res
+
+def _add_extant_column_to_spec(extant_cols, column_name, res, column_type, num, description):
```

Comment:
I'd add a \n + some spaces between `spec(` and `extant_cols` to get this below 80 chars (ignore these comments if it's under 80!)